templates/vsphere/cluster-template.yaml

```diff
@@ -0,0 +1,1514 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  name: ${CLUSTER_NAME}-vsphere-proxy-config
+stringData:
+  proxy-config: |
+    ${PROXY_DISABLE}[Service]
+    ${PROXY_DISABLE}Environment="HTTPS_PROXY=http://squid.internal:3128"
+    ${PROXY_DISABLE}Environment="HTTP_PROXY=http://squid.internal:3128"
+    ${PROXY_DISABLE}Environment="https_proxy=http://squid.internal:3128"
+    ${PROXY_DISABLE}Environment="http_proxy=http://squid.internal:3128"
+    ${PROXY_DISABLE}Environment="NO_PROXY=10.0.0.0/8,192.168.0.0/16,127.0.0.1,localhost,0.0.0.0"
+    ${PROXY_DISABLE}Environment="no_proxy=10.0.0.0/8,192.168.0.0/16,127.0.0.1,localhost,0.0.0.0"
```

Comment:
we should not hardcode this. Consider whether we could have an option like `PROXY_CONFIGURATION` with a value like

```
PROXY_CONFIGURATION="
http_proxy=...
https_proxy=...
no_proxy=....
"
```

Then use `[Service] EnvironmentFile` and point to that file instead.

Reply:
See https://www.freedesktop.org/software/systemd/man/latest/systemd.service.html#Command%20lines for the `EnvironmentFile` directive